src/gp-model/main.py

```diff
@@ -206,39 +206,48 @@ def predict_model(m, tenant, airqloud):
 
     return result
 
-def periodic_function(tenant, airqloud):
+def periodic_function(tenant, airqloud, aq_id):
     '''
     Re-trains the model regularly
     '''
-    print('starting')
     X = np.zeros([0,3])
     Y = np.zeros([0,1])
-    channels = get_channels_ts(airqloud)
-    print('ongoing')
-    for channel in channels:
-        d = download_seven_days_ts(channel['id'], channel['api_key'])
-        if d.shape[0]!=0:
-            d = preprocessing_ts(d)
-            df = pd.DataFrame({'channel_id':[channel['id']], 
-                                'longitude':[channel['long']], 
-                                'latitude':[channel['lat']]})
-        
-            Xchan = np.c_[np.repeat(np.array(df)[:,1:],d.shape[0],0),[n.timestamp()/3600 for n in d['created_at']]]
-            Ychan = np.array(d['field1'])
-            X = np.r_[X,Xchan]
-            Y = np.r_[Y,Ychan[:, None]]
-    m = train_model(X, Y, airqloud)
-    predict_model(m, tenant, airqloud)
+    
+    poly, min_long, max_long, min_lat, max_lat = get_airqloud_polygon(tenant, airqloud)
+    devices = get_devices_in_airqloud(poly, tenant)
+    if len(devices)>0:
+        for device in devices:
+            df = get_pm_data(device['name'], device['latitude'], device['longitude'], tenant)
+            if df.shape[0]!=0:
+                prep_df = preprocessing(df)
+                Xchan = np.asarray(prep_df.iloc[:, :3])
+                Ychan = np.asarray(prep_df.iloc[:, -1])
+                X = np.r_[X,Xchan]
+                Y = np.r_[Y,Ychan[:, None]]
+        m = train_model(X, Y, airqloud)
+        predict_model(m, tenant, airqloud, aq_id, poly, min_long, max_long, min_lat, max_lat)
+    else:
+        pass
 
+def get_all_airqlouds(tenant):
+    '''
+    Returns a list of all the airqlouds for a particuar tenant
+    '''
+    params = {'tenant':tenant}
+    airqlouds = requests.get(VIEW_AIRQLOUD_URI, params=params).json()['airqlouds']
+    names = [aq['name'] for aq in airqlouds]
+    aq_ids = [aq['_id'] for aq in airqlouds]
+    return names, aq_ids
 
 if __name__=='__main__':
+    airqloud_names, aq_ids = get_all_airqlouds('airqo')
```

Comment:
Thanks @lilly-muyama 
How shall we switch to the airqlouds of other orgs/tenants

Reply:
@Baalmart AirQlouds are an internal organization thing. Otherwise, if that situation ever arises, it should be easily resolved.

Reply:
Alright, sounds good. Thanks. I guess we are already have AirQlouds for KCCA in staging. @lilly-muyama